### PR TITLE
Update component phase inputs to require both phase and completion date if phase simple is "Complete"

### PR DESCRIPTION
**Pull request by mddilley**
## Associated issues
Closes https://github.com/cityofaustin/atd-data-tech/issues/19113

This PR updates the component form validation and behavior so that:
- Completion Date is required if the Phase is **Complete** or **Post-Construction** (both have a simple name of Complete - see `moped_phases` table for reference)
- Completion Date is only available to be added if the Phase is **Complete** or **Post-Construction**

## Testing
**URL to test:** 

https://deploy-preview-1430--atd-moped-main.netlify.app/

**Steps to test:**
1. Go to an existing project and create a new component in the map tab
2. While using the create form, toggle the **Use component phase** switch on and choose any phase except for **Complete** or **Post-Construction**
3. You should see that the **Completion Date** field is now hidden with any phase except for **Complete** or **Post-Construction**
4. Now, choose **Complete** or **Post-Construction** and notice that a validation error requires you to enter a **Completion Date** if you have chosen either
5. Save your component and try editing the phase and completion date too
6. Try out different orders of adding **Complete** and then a **Completion Date** and then clearing the date or phase to make sure that the form **Continue** (creating) or **Save** (editing) is active when the validation requirements are met.

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Product manager checked [DB view dependencies](https://atd-dts.gitbook.io/moped-documentation/product-management/updating-the-arcgis-online-components-database-view)
- [x] Product manager added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/queries/components.js
Comment (mddilley) [inline moped-editor/src/queries/components.js:1]: The updates are so we fetch the `phase_name_simple` when getting phase options for the autocomplete (creating) and when we fetch the existing phase of a component (editing).

File: moped-editor/src/views/projects/projectView/DateFieldEditComponent.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/DateFieldEditComponent.js:16]: This new prop is added so we can pass error state and error message to the TextField within the MUI MobileDatePicker. The validation error state and messages provided by React Hook Form/Yup are passed in the component form.

File: moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js:157]: This clears completion date if the phase is changed so we don't leave a completion date hanging around in the form if there shouldn't be one.

File: moped-editor/src/views/projects/projectView/ProjectComponents/utils/form.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponents/utils/form.js:397]: This was a TIL for me. when we clear the completion date programmatically when the phase option is changed, the validation is triggered. This has the same effect as when any input in the form changes by a user (`mode: "onChange"` in the config for `useForm`).

See https://react-hook-form.com/docs/useform/setvalue

**Comment by tillyw** [comment]
Screenshot 2024-09-25 at 6 04 18 PM
should the completion date field go away when you clear the phase or change to a phase that does not require a completion date?

**Review by tillyw — APPROVED** [review]
code looks good, tested and works well! aside from the question i asked about hiding the field when changing the phase this has my stamp of approval!

**Comment by mddilley** [comment]
@tillyw that is a bug in your screenshot. I'll fix. Thanks for catching! 🙏

**Review by chiaberry — APPROVED** [review]
this is a great upgrade for our data quality 🚢

File: moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js:70]: 🔥 yup is so great and so are you for doing cool things with it 👍

**Review by johnclary — APPROVED** [review]
This is a nice little upgrade—thanks, Mike!

two minor things:

1. If the date input supports the "small" size, it would be nice to slim it down to match the other form inputs
2. Just mentioning again that the autocomplete is barking a warning about the selected value not matching any of the available options

here's the form input where you can see the size difference

Screenshot 2024-09-26 at 10 53 12 AM


🚢 🚢 🚢 🚢

**Comment by mddilley** [comment]
@johnclary thanks for flagging the autocomplete warning. i saw the same in chia's PR so I'm going to try to fix in here before merge. I'll take a look at that completion date width too. I'm wondering if the error message it extending it when shown. 🔍

**Comment by mddilley** [comment]
I ran out of time and thus a new snackoo is born for the phase option warning. 🐣 🍫  https://github.com/cityofaustin/atd-data-tech/issues/19221

File: moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js
Comment (johnclary) [reply moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js:70]: i've been wanting to see what the fuss is about. Yup does typescript stuff too, fwiw, but it also seems like it's just barely maintained by one person. Zod looks great.

**Pull request merged**
